Vault/Library/CachePersistence.cs

```diff
@@ -124,10 +128,93 @@ void AfterAccessNotification(AuthenticationRecord authRecord)
                 // Call AuthenticateAsync to fetch a new AuthenticationRecord.
                 authRecord = credential.Authenticate();
 
+                // Create the original data to be encrypted
+                var toEncrypt = Encoding.ASCII.GetBytes(authRecord.ToString());
+
+                // Create some random entropy.
+                byte[] entropy = CreateRandomEntropy();
+
                 // Serialize the AuthenticationRecord to disk so that it can be re-used across executions of this initialization code.
-                var authRecordStream = new FileStream(FileName, FileMode.Create, FileAccess.Write);
-                authRecord.SerializeAsync(authRecordStream);
+                var authRecordStream = new FileStream(FileName, FileMode.OpenOrCreate);
+
+                // Encrypt a copy of the data to the stream.
+                int bytesWritten = EncryptDataToStream(toEncrypt, entropy, DataProtectionScope.CurrentUser, authRecordStream);
+
+                authRecordStream.Close();
             }
         }
+
+        public static byte[] CreateRandomEntropy()
+        {
+            // Create a byte array to hold the random value.
+            byte[] entropy = new byte[16];
+
+            // Create a new instance of the RNGCryptoServiceProvider.
+            // Fill the array with a random value.
+            new RNGCryptoServiceProvider().GetBytes(entropy);

```

Comment:
This is disposable so you should use something like `using var rng = new RNGCryptoServicePRovider();` You could also save some time by declaring this as a `static readonly` field instead of initializing it with every call.